lib/membrane_rtc_engine/endpoints/webrtc/variant_selector.ex

```diff
@@ -123,27 +224,27 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.VariantSelector do
 
       selector = %__MODULE__{selector | target_variant: variant}
 
-      {selector, nil}
+      {selector, :noop}
     end
   end
 
-  defp select_variant(%__MODULE__{active_variants: variants} = selector) do
-    variant = best_active_variant(variants)
+  defp select_variant(%__MODULE__{} = selector) do
+    variant = best_active_variant(selector)
     select_variant(selector, variant)
   end
 
-  defp select_variant(selector, nil) do
+  defp select_variant(selector, :no_variant) do
     Membrane.Logger.debug("No active variant.")
-    selector = %__MODULE__{selector | current_variant: nil, queued_variant: nil}
-    {selector, nil}
+    selector = %__MODULE__{selector | current_variant: :no_variant, queued_variant: nil}
+    {selector, :stop}
```

Comment:
That's not fully correct. We should return `:stop` only when we are no longer able to send track to the browser because of bandwidth limitation.

E.g. we can end up here when the last variant we were using became inactive. In such a case `noop` should be returned. 

Or we got our initial allocation but the first variant that became active was :high. In such a case there is nothing to stop.

I am not sure if stop should be handled in this function. Maybe it should be handled one level higher i.e. only in set_current_bandwidth :thinking:

Reply:
As discussed, it will be fixed in another PR